### PR TITLE
Upstream pull followup - Reuse upstream propagation code

### PR DESCRIPTION
**Pull request by dszmigielski**
## Why

We don't need our propagation code anymore as it is now available in the upstream repo.

## What

Reuse upstream propagation code which was recently refactored here:  [SpanContextPropagator refactor by tonyredondo · Pull Request #2536 · DataDog/dd-trace-dotnet](https://github.com/DataDog/dd-trace-dotnet/pull/2536)

### CODE REVIEW

File: tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs [hunk 1]
```diff
@@ -376,6 +368,16 @@ internal static partial class ConfigurationKeys
         /// <seealso cref="TracerSettings.ExpandRouteTemplatesEnabled"/>
         public const string ExpandRouteTemplatesEnabled = "SIGNALFX_TRACE_EXPAND_ROUTE_TEMPLATES_ENABLED";
 
+        /// <summary>
+        /// Configuration key for setting the propagation style injection.
+        /// </summary>
+        public const string PropagationStyleInject = "SINGALFX_PROPAGATION_STYLE_INJECT";
```
Comment (RassK) [inline tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs:374]: this is a breaking change. not sure if we should actually do it.
Might be better to leave the original `SIGNALFX_PROPAGATORS`, document these 2 new vars in internal config and assign value from `SIGNALFX_PROPAGATORS`

Comment (dszmigielski) [reply tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs:374]: Good point, I will preserve our variable name and populate new properties from its value.

**Review by pjanotti — CHANGES_REQUESTED** [review]
I didn't look at it yet, but, let's block merges until pull from upstream is done.

**Comment by Kielek** [comment]
Do you see any option to remove everything under `Datadog.Trace.Propagation` namespace? It is not available on the upstream side.

**Comment by dszmigielski** [comment]
> Do you see any option to remove everything under `Datadog.Trace.Propagation` namespace? It is not available on the upstream side.

What is left there are header names and some extensions that we are still using. I would rather leave them be tbh.

**Review by RassK — APPROVED** [review]

File: tracer/src/Datadog.Trace/Configuration/TracerSettings.cs [hunk 2]
```diff
@@ -188,6 +186,10 @@ public TracerSettings(IConfigurationSource source)
             DelayWcfInstrumentationEnabled = source?.GetBool(ConfigurationKeys.FeatureFlags.DelayWcfInstrumentationEnabled)
                                             ?? false;
 
+            PropagationStyleInject = TrimSplitString(source?.GetString(ConfigurationKeys.Propagators) ?? nameof(Propagators.ContextPropagators.Names.B3), ',').ToArray();
```
Comment (Kielek) [inline tracer/src/Datadog.Trace/Configuration/TracerSettings.cs:189]: You have the same logic twice. Consider to call it only once.

File: tracer/test/snapshots/GrpcLegacy.SubmitTraces_httpclient=Disabled.verified.txt [hunk 3]
```diff
@@ -2,64 +2,77 @@
   {
     TraceId: Id_1,
     SpanId: Id_2,
+    Name: SendBothStreamingRequest,
```
Comment (Kielek) [inline tracer/test/snapshots/GrpcLegacy.SubmitTraces_httpclient=Disabled.verified.txt:5]: I am not able to verify this file. Please check if the changes are expected.

**Review by Kielek — COMMENTED** [review]
One more thing, did you have chance to verify if long TraceId is correctly propagated? As I remember upstream supports only short TraceIds.

**Pull request merged**